### PR TITLE
Enforce access controls to getWorkspace, and add error messaging.

### PR DESCRIPTION
**Pull request by blrubenstein**
This enforces access to the getWorkspace call based on access to Firecloud, by making a call to the FC workspace from our own `getWorkspace` call. Step two will be requiring write access in order to update the workspace in any way.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java [hunk 1]
```diff
@@ -98,12 +101,62 @@ public Workspace apply(org.pmiops.workbench.db.model.Workspace workspace) {
...
+                  .researchPurpose(researchPurpose);
+              if (fcWorkspace.getCreatedBy() != null) {
+                result.setCreator(fcWorkspace.getCreatedBy());
+              }
+              if (workspace.getCdrVersion() != null) {
+                result.setCdrVersionId(String.valueOf(workspace.getCdrVersion().getCdrVersionId()));
+              }
+
+
+              result.setUserRoles(workspace.getWorkspaceUserRoles().stream().map(TO_CLIENT_USER_ROLE).collect(Collectors.toList()));
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java]: Should we use fcWorkspace for the roles?

File: api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java [hunk 2]
```diff
@@ -280,7 +333,19 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
   public ResponseEntity<Workspace> getWorkspace(String workspaceNamespace, String workspaceId) {
     org.pmiops.workbench.db.model.Workspace dbWorkspace = workspaceService.getRequired(
         workspaceNamespace, workspaceId);
-    return ResponseEntity.ok(TO_CLIENT_WORKSPACE.apply(dbWorkspace));
+    org.pmiops.workbench.firecloud.model.Workspace fcWorkspace;
+    try {
+      fcWorkspace = fireCloudService.getWorkspace(
+          workspaceNamespace, workspaceId).getWorkspace();
+    } catch (org.pmiops.workbench.firecloud.ApiException e) {
+      if (e.getCode() == 404) {
```
Comment (danqrodney) [inline api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java]: Is a 404 returned if you don't have access to a workspace?

If instead it's a 403, we should probably handle that here?

Comment (blrubenstein) [reply api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java]: It is a 404 from FC.

File: api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java (hunk 1 above)
Comment (blrubenstein) [reply api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java]: You don't actually get the roles from getWorkspace in FC, just your own role. I am planning on using that for enforcing write access or higher.

Comment (danqrodney) [reply api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java]: SGTM. Thanks!

**Pull request merged**